diff --git a/GNUmakefile.in b/GNUmakefile.in
index 68bcc1c..2a63213 100644
--- a/GNUmakefile.in
+++ b/GNUmakefile.in
@@ -9214,7 +9214,9 @@ libWebCoreJS_depfiles := $(patsubst \
 webinspectordir = ${datadir}/webkit-1.0/webinspector
 dist_webinspector_DATA = \
 	$(WebCore)/English.lproj/localizedStrings.js \
-	$(shell ls $(WebCore)/inspector/front-end/*.{js,html,css})
+	$(shell ls $(WebCore)/inspector/front-end/*.js) \
+	$(shell ls $(WebCore)/inspector/front-end/*.html) \
+	$(shell ls $(WebCore)/inspector/front-end/*.css)
 
 webinspectorimagesdir = ${datadir}/webkit-1.0/webinspector/Images
 dist_webinspectorimages_DATA = \
diff --git a/WebCore/GNUmakefile.am b/WebCore/GNUmakefile.am
index 416f00c..fb45261 100644
--- a/WebCore/GNUmakefile.am
+++ b/WebCore/GNUmakefile.am
@@ -3768,7 +3768,9 @@ webcore_dist += \
 webinspectordir = ${datadir}/webkit-1.0/webinspector
 dist_webinspector_DATA = \
 	$(WebCore)/English.lproj/localizedStrings.js \
-	$(shell ls $(WebCore)/inspector/front-end/*.{js,html,css})
+	$(shell ls $(WebCore)/inspector/front-end/*.js) \
+	$(shell ls $(WebCore)/inspector/front-end/*.html) \
+	$(shell ls $(WebCore)/inspector/front-end/*.css)
 
 webinspectorimagesdir = ${datadir}/webkit-1.0/webinspector/Images
 dist_webinspectorimages_DATA = \
