Index: webkitgtk/Source/WebKit/gtk/webkit/webkitwebview.cpp
===================================================================
--- webkitgtk.orig/Source/WebKit/gtk/webkit/webkitwebview.cpp	2013-09-16 15:23:19.214982057 -0300
+++ webkitgtk/Source/WebKit/gtk/webkit/webkitwebview.cpp	2013-09-16 15:23:19.210982038 -0300
@@ -789,6 +789,12 @@
     if (!frame.view())
         return FALSE;
 
+#ifndef GTK_API_VERSION_2
+    /* Ignore Alt-scroll events in GTK3 so that GtkNotebook can use it to switch tabs */
+    if (event->state & GDK_MOD1_MASK)
+        return FALSE;
+#endif
+
     PlatformWheelEvent wheelEvent(event);
     return frame.eventHandler().handleWheelEvent(wheelEvent);
 }
