Index: webkitgtk-experimental/Source/WebKit/gtk/webkit/webkitwebview.cpp
===================================================================
--- webkitgtk-experimental.orig/Source/WebKit/gtk/webkit/webkitwebview.cpp	2013-02-23 22:04:16.497835771 -0300
+++ webkitgtk-experimental/Source/WebKit/gtk/webkit/webkitwebview.cpp	2013-02-23 22:04:16.493835703 -0300
@@ -800,6 +800,12 @@
     if (!frame->view())
         return FALSE;
 
+#ifndef GTK_API_VERSION_2
+    /* Ignore Alt-scroll events in GTK3 so that GtkNotebook can use it to switch tabs */
+    if (event->state & GDK_MOD1_MASK)
+        return FALSE;
+#endif
+
     PlatformWheelEvent wheelEvent(event);
     return frame->eventHandler()->handleWheelEvent(wheelEvent);
 }
