Description: Fix wide pointer issues on ia64 (closes: #642750).
Author: Stephan Schreiber <info@fs-driver.org>
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -688,6 +688,13 @@
 #define ENABLE_JIT 0
 #endif
 
+/* FIXME: The fast malloc implementation is broken on Itanium / IA64 because
+   some memory barriers are missing in the thread-unsafe code around the
+   pagemap_cache_ object. */
+#if CPU(IA64) || CPU(IA64_32)
+#define USE_SYSTEM_MALLOC 1
+#endif
+
 /* The JIT is enabled by default on all x86, x86-64, ARM & MIPS platforms. */
 #if !defined(ENABLE_JIT) \
     && (CPU(X86) || CPU(X86_64) || CPU(ARM) || CPU(ARM64) || CPU(MIPS)) \
