--- a/GNUmakefile.am
+++ b/GNUmakefile.am
@@ -49,6 +49,7 @@
 bin_PROGRAMS :=
 noinst_PROGRAMS :=
 libexec_PROGRAMS :=
+noinst_DATA :=
 noinst_HEADERS :=
 noinst_LTLIBRARIES :=
 lib_LTLIBRARIES :=
--- a/Source/WebCore/GNUmakefile.am
+++ b/Source/WebCore/GNUmakefile.am
@@ -734,7 +734,7 @@
 
 # It seems that $(shell) does not expand when it is a rule dependency, so
 # we must redefine this list of copied files with traditional dependency wildcards.
-noinst_DATA = ${GENSOURCES_INSPECTOR}/inspector.html
+noinst_DATA += ${GENSOURCES_INSPECTOR}/inspector.html
 ${GENSOURCES_INSPECTOR}/inspector.html: $(WebCore)/inspector/front-end/*.html \
 		$(WebCore)/inspector/front-end/*.js \
 		$(WebCore)/inspector/front-end/UglifyJS/*.js \
--- a/Tools/GNUmakefile.am
+++ b/Tools/GNUmakefile.am
@@ -417,6 +417,6 @@
 endif
 
 if ENABLE_GTK_DOC
-all: docs-build.stamp
+noinst_DATA += docs-build.stamp
 endif
 
