--- trunk/Source/WebKit/gtk/webkit/webkitwebnavigationaction.cpp	(revision 147556)
+++ trunk/Source/WebKit/gtk/webkit/webkitwebnavigationaction.cpp	(revision 147557)
@@ -135,5 +135,5 @@
     objectClass->get_property = webkit_web_navigation_action_get_property;
     objectClass->set_property = webkit_web_navigation_action_set_property;
-    objectClass->dispose = webkit_web_navigation_action_finalize;
+    objectClass->finalize = webkit_web_navigation_action_finalize;
 
     /**
