From: Alberto Garcia <berto@igalia.com>
Subject: Install all auxiliary binaries in the same directory.
Bug: https://bugs.webkit.org/show_bug.cgi?id=136209
Index: webkitgtk/Source/cmake/OptionsGTK.cmake
===================================================================
--- webkitgtk.orig/Source/cmake/OptionsGTK.cmake
+++ webkitgtk/Source/cmake/OptionsGTK.cmake
@@ -20,7 +20,7 @@ set(ENABLE_WAYLAND_TARGET OFF CACHE BOOL
 # CMAKE_INSTALL_* variables that are populated by the GNUInstallDirs macro.
 set(LIB_INSTALL_DIR "${CMAKE_INSTALL_FULL_LIBDIR}" CACHE PATH "Absolute path to library installation directory")
 set(EXEC_INSTALL_DIR "${CMAKE_INSTALL_FULL_BINDIR}" CACHE PATH "Absolute path to executable installation directory")
-set(LIBEXEC_INSTALL_DIR "${CMAKE_INSTALL_FULL_LIBEXECDIR}/webkitgtk-${WEBKITGTK_API_VERSION}" CACHE PATH "Absolute path to install executables executed by the library")
+set(LIBEXEC_INSTALL_DIR "${CMAKE_INSTALL_FULL_LIBEXECDIR}/webkit2gtk-${WEBKITGTK_API_VERSION}" CACHE PATH "Absolute path to install executables executed by the library")
 
 set(DATA_BUILD_DIR "${CMAKE_BINARY_DIR}/share/${WebKit_OUTPUT_NAME}")
 set(DATA_INSTALL_DIR "${CMAKE_INSTALL_DATADIR}/webkitgtk-${WEBKITGTK_API_VERSION}")
