From: Yusuke Suzuki <utatane.tea@gmail.com>
Subject: Fix FTBFS in PowerPC
Bug: https://bugs.webkit.org/show_bug.cgi?id=156015
Origin: https://github.com/WebKit/webkit/commit/a7cdcdc77b8efef239d9b06e857a64e490d1a81b
Index: webkitgtk/Source/JavaScriptCore/heap/MachineStackMarker.cpp
===================================================================
--- webkitgtk.orig/Source/JavaScriptCore/heap/MachineStackMarker.cpp
+++ webkitgtk/Source/JavaScriptCore/heap/MachineStackMarker.cpp
@@ -86,7 +86,11 @@ static void pthreadSignalHandlerSuspendR
     }
 
     ucontext_t* userContext = static_cast<ucontext_t*>(ucontext);
+#if CPU(PPC)
+    thread->suspendedMachineContext = *userContext->uc_mcontext.uc_regs;
+#else
     thread->suspendedMachineContext = userContext->uc_mcontext;
+#endif
 
     // Allow suspend caller to see that this thread is suspended.
     // sem_post is async-signal-safe function. It means that we can call this from a signal handler.
