From: Khem Raj <raj.khem@gmail.com>
Subject: Fix FTBFS with GCC 5 in some architectures
Bug: https://bugs.webkit.org/show_bug.cgi?id=147815
Index: webkitgtk/Source/JavaScriptCore/runtime/JSObject.cpp
===================================================================
--- webkitgtk.orig/Source/JavaScriptCore/runtime/JSObject.cpp
+++ webkitgtk/Source/JavaScriptCore/runtime/JSObject.cpp
@@ -1965,6 +1965,11 @@ void JSObject::putByIndexBeyondVectorLen
     }
 }
 
+// Explicit instantiations needed by JSArray.cpp.
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<Int32Shape>(ExecState* exec, unsigned i, JSValue value);
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<DoubleShape>(ExecState* exec, unsigned i, JSValue value);
+template void JSObject::putByIndexBeyondVectorLengthWithoutAttributes<ContiguousShape>(ExecState* exec, unsigned i, JSValue value);
+
 void JSObject::putByIndexBeyondVectorLengthWithArrayStorage(ExecState* exec, unsigned i, JSValue value, bool shouldThrow, ArrayStorage* storage)
 {
     VM& vm = exec->vm();
