From: Alberto Garcia <berto@igalia.com>
Subject: Fix build failure on x86 due to incorrect asm generation
Bug: https://bugs.webkit.org/show_bug.cgi?id=126985
Index: webkit/Source/WTF/wtf/Atomics.h
===================================================================
--- webkit.orig/Source/WTF/wtf/Atomics.h
+++ webkit/Source/WTF/wtf/Atomics.h
@@ -295,7 +295,7 @@ inline bool weakCompareAndSwap(uint8_t*
         "lock; cmpxchgb %3, %2\n\t"
         "sete %1"
         : "+a"(expected), "=q"(result), "+m"(*location)
-        : "r"(newValue)
+        : "q"(newValue)
         : "memory"
         );
     return result;
