From: Alberto Garcia <berto@igalia.com>
Subject: Fix build failure on x86 due to incorrect asm generation
Bug: https://bugs.webkit.org/show_bug.cgi?id=126985
Index: webkit/Source/WTF/wtf/Atomics.h
===================================================================
--- webkit.orig/Source/WTF/wtf/Atomics.h	2014-01-14 18:06:16.750623136 +0000
+++ webkit/Source/WTF/wtf/Atomics.h	2014-01-15 10:38:11.456046548 +0000
@@ -289,7 +289,7 @@
 inline bool weakCompareAndSwap(uint8_t* location, uint8_t expected, uint8_t newValue)
 {
 #if ENABLE(COMPARE_AND_SWAP)
-#if !OS(WINDOWS) && (CPU(X86) || CPU(X86_64))
+#if !OS(WINDOWS) && CPU(X86_64)
     unsigned char result;
     asm volatile(
         "lock; cmpxchgb %3, %2\n\t"
